modules/tinymce/src/core/main/ts/fmt/ListItemFormat.ts

```diff
@@ -32,25 +32,44 @@ const isRngEndAtEndOfElement = (rng: Range, elm: Element) => {
 
 const isEditableListItem = (dom: DOMUtils) => (elm: Element) => NodeType.isListItem(elm) && dom.isEditable(elm);
 
-const getFullySelectedBlocks = (selection: EditorSelection) => {
-  const blocks = selection.getSelectedBlocks();
-  const rng = selection.getRng();
+// TINY-13197: If the content is wrapped inside a block element, the first block returned by getSelectedBlocks() is not LI, even when the content is fully selected.
+// However, the second and subsequent do return LI as the selected block so only the first block needs to be adjusted
+const normalizeFirstBlockIf = (selection: EditorSelection, blocks: Element[], pred: (block: Element) => boolean) =>
+  Arr.map(blocks, (block, i) => {
+    if (i === 0 && pred(block)) {
+      return Optional.from(selection.dom.getParent(block, NodeType.isListItem)).getOr(block);
```

Comment:
I think I'm going to start suggesting this more often in response to `Optional.from(...).getOr()`​. Optionals still have their place but the language has fine tools for this now.

```suggestion
      return selection.dom.getParent(block, NodeType.isListItem) ?? block;
```